src/one_hot_encoding.jl

```diff
@@ -0,0 +1,41 @@
+"""
+    OneHotEncoding <: Transform
+
+One-hot encode the categorical value for each target element.
+
+Construct a n-by-p binary matrix, given a `Vector` of target data `x` (of length n) and a
+`Vector` of all unique possible values in x (of length p).
+
+The element [i, j] is `1` if the i^th target in `x` corresponds to the j^th possible
+value and `0` otherwise.
+
+Note that this Transform does not support specifying dims other than `:` (all dims) because
+it is a one-to-many transform (for example a `Vector` input produces a `Matrix` output).
+"""
+struct OneHotEncoding <: Transform
+    categories::Dict{Any, Int}
+
+    function OneHotEncoding(possible_values::AbstractVector)
+        if length(unique(possible_values)) < length(possible_values)
+            throw(ArgumentError("Expected an ordered list of all unique possible values"))
```

Comment:
Do the elements need to be a consistent order? Or does "ordered list" just refer to the type?

Reply:
They just need to be in the order of expected column output. so for input of 24 hours they should be ordered from Hour(1) to Hour(24). Then if it is applied on x = [Hour (2), Hour(2), Hour(1)] the result would be

```
 0  1  0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0  1  0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 1  0  0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 ```